File: ImagePickerMaster/src/main/java/com/example/apple/glidetest/bean/Folder.java
package com.example.apple.glidetest.bean;

import android.text.TextUtils;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Apple on 17/5/26.
 */

public class Folder {

    public String dir;
    public String name;
    public String firstImagePath;
    public List<String> imgs = new ArrayList<>();

    public Folder(String dir, String name) {
        this.dir = dir;
        this.name = name;
    }

    public Folder(String dir, String name, String firstImagePath) {
        this.dir = dir;
        this.name = name;
        this.firstImagePath = firstImagePath;
    }

    public void addImage(String path) {
        if (!TextUtils.isEmpty(path)) {
            imgs.add(path);
        }
    }

    public int getCount() {
        return imgs == null ? 0 : imgs.size();
    }
}

File: ImagePickerMaster/src/main/java/com/example/apple/glidetest/bean/Folder.kt
package com.example.apple.glidetest.bean

import java.util.*

/**
 * Created by Apple on 17/5/26.
 */

class Folder(var dir: String, var name: String) {

    var firstMedia: Media? = null
    var medias = ArrayList<Media>()

    constructor(dir: String, name: String, firstMedia: Media) : this(dir, name) {
        this.firstMedia = firstMedia
    }

    fun addMedia(media: Media, index:Int?=null) {
        if (!medias.contains(media)) {
//            java.lang.ArrayIndexOutOfBoundsException: length=10; index=10  java.lang.ArrayIndexOutOfBoundsException: length=15; index=15
            if (index!=null && index >= 0) medias.add(index,media)
            else medias.add(media)
        }
    }

    fun sort(){
        Collections.sort(medias)
    }

    var count: Int = 0
        get() {return medias.size}

    fun clear(){
        dir = ""
        name =""
        firstMedia = null
        medias.clear()
    }
}
